src/QsCompiler/TextProcessor/CompletionParsing.fs

```diff
@@ -5,49 +5,110 @@
 /// closed if it is still valid for the bracket to be closed later.
 module Microsoft.Quantum.QsCompiler.TextProcessing.CompletionParsing
 
+#nowarn "40"
```

Comment:
This is to disable the warning F# gives for defining recursive values - I'm defining some parsers recursively (using the `parser` computation expression to delay evaluation and avoid an infinite loop) since IMO it's slightly more readable than using `createParserForwardedToRef`. I haven't noticed any performance problems but I can change the recursive parsers to use `createParserForwardedToRef` if it's an issue.

Reply:
Ok, let's leave it for now if perf is fine, since you only ever parse a very small piece of text and the requirements in that regard are not likely to change.